merkledag/node.go

```diff
@@ -76,6 +76,11 @@ func MakeLink(n *Node) (*Link, error) {
 	}, nil
 }
 
+// GetCachedNode returns the MDAG Node that was cached, or nil
+func (l *Link) GetCachedNode() *Node {
```

Comment:
I think we can just do without caching entirely, and actually go all the way and remove the `link.Node` field.